src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -121,7 +122,9 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
 
     DockerBuildImage() {
         inputDir.set(project.file('docker'))
-        File imageIdTextFile = new File(project.buildDir, ".docker/${path}-imageId.txt")
+
+        String safeTaskPath = path.replaceFirst("^:", "").replaceAll(":", "_")
```

Comment:
Thanks for the PR. Maybe we should just hash the task path. Might make it easier to avoid other issues in the future with Windows.

Reply:
eehhh ... I would say lets wait until something arises to do that IMO. This is simple and straightforward and folks can easily digest what is/is-not going on fairly easy just be looking at files on disk.

Reply:
Happy to keep it as is. I don't think using Base64 would me more complex and could be decoded if needed.

```
Base64.getEncoder().encodeToString(path.getBytes("utf-8"))
```

Reply:
The [`Base64` class](https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html) is just plain Java 8+.

Reply:
Didn't even know they had that in java8 which goes to show how long its been since I've needed to do this :)